tests/library/test_pydantic.py

```diff
@@ -0,0 +1,174 @@
+import inspect
+from json import dumps as json_dumps
+from typing import Any, List, Literal, Type, Union
+
+import pydantic
+import pytest
+from pydantic.json_schema import to_jsonable_python
+
+from guidance import json as gen_json
+from guidance import models
+from guidance import pydantic as gen_pydantic
+from guidance._parser import ParserException
+
+
+def to_compact_json(target: Any) -> str:
```

Comment:
Isn't this just the same as the one in `library/_json.py`?

Reply:
Very nearly. Note the `default` arg to `json_dumps` here -- it makes sure that the input can be dumped even in the case that it's not a standard jsonable python object (basically if it hits a pydantic object somwhere, it'll automatically call `obj.model_dump_json`. I considered adding this to the implementation in `_json`, but I thought keeping all of the "pydantic stuff" separate was sensible. Would you recommend (1) just putting this in `_json` or (2) renaming this something more descriptive?